addons/mail/static/src/model/model_core.js

```diff
@@ -341,16 +348,29 @@ export function patchRecordMethods(modelName, recordMethods) {
 }
 
 /**
- * @param {Object} definition The JSON definition of the model to register.
- * @param {Object} [definition.fields]
- * @param {string[]} definition.identifyingFields
- * @param {Object} [definition.lifecycleHooks]
- * @param {Object} [definition.modelGetters] Deprecated; use fields instead.
- * @param {Object} [definition.modelMethods]
- * @param {string} definition.name
- * @param {OnChanges[]} [definition.onChanges]
- * @param {Object} [definition.recordGetters] Deprecated; use fields instead.
- * @param {Object} [definition.recordMethods]
+ * @typedef {Object} ModelDefinition
+ * @property {Object} [fields] Fields of the model.
+ * Keys are the name of the field, value are the field definition
+ * @property {Array<string|string[]>} identifyingFields Fields used to uniquely identify a record of this model.
+ * - [field1, fields2] means field1 AND field2
+ * - [[field1], [field2]] means field1 OR field2.
```

Comment:
This is AND not OR, the OR part is just one element in each case here. But it's a dummy example that we don't want in practice, probably remove it or replace it with an actual OR example.

Reply:
For OR yes this is the syntax.